IsogeomGenerator/generate_isogeom.py

```diff
@@ -0,0 +1,332 @@
+                        action='store_true',
+                        required=False,
+                        dest='tagviz',
+                        help='If set, surfaces generated will be tagged ' +
+                        'with data for visualization purposes.'
+                        )
+    parser.add_argument('-g', '--geomfile',
+                        action='store',
+                        nargs=1,
+                        required=False,
+                        default=None,
+                        metavar='GEOM_FILENAME',
+                        dest='geomfile',
+                        type=str,
+                        help='Filename to write generated isosurface ' +
+                        'geometry file. ' +
+                        'Must be either a .h5m or .vtk file name. ' +
+                        'Default name is isogeom.h5m.'
+                        )
+    parser.add_argument('-sp', '--savepath',
+                        action='store',
+                        nargs=1,
+                        required=False,
+                        default=None,
+                        metavar='PATH',
+                        dest='savepath',
+                        type=str,
+                        help='Absolue path to folder to write generated ' +
+                        'geometry file. If not set, file will be saved in ' +
+                        'the database folder (-db).'
+                        )
+    parser.add_argument('-t', '--tag',
+                        action='append',
+                        nargs=2,
+                        required=False,
+                        default=None,
+                        metavar='TAGNAME TAGVAL',
+                        dest='tags',
+                        help='Information to tag on the whole geometry. ' +
+                        'First entry must be the name for the tag (string). ' +
+                        'Second entry must be the value for the tag (will ' +
+                        'be tagged as float). ' +
+                        'Option can be set more than once to set more tags.'
+                        )
+
+
+def set_shared_options(parser, moab=False):
+    """set options that are in both the moab and visit steps.
+    """
+    set_level_options(parser, moab)
+    parser.add_argument('-db', '--database',
+                        action='store',
+                        nargs=1,
+                        required=False,
+                        default="/tmp",
+                        metavar='DATABASE_PATH',
+                        dest='db',
+                        type=str,
+                        help='Relative path to folder where isosurface ' +
+                        'meshfiles from VisIt are located or to be saved. ' +
+                        'Default is a folder named tmp/ in the current ' +
+                        'directory.'
+                        )
+
+
+def parse_arguments():
+    """parse user args
+    """
```

Comment:
I am wondering if having a full map of all the parsing option there would be usefull

Reply:
Having this visualization/categorization in `README.md` would be great.

Reply:
I'll add a table of usage information to the README